integration-test/src/test/java/org/apache/iotdb/db/it/aggregation/IoTDBCountIfIT.java

```diff
@@ -0,0 +1,181 @@
+          "Count_if(root.db.d1.s1 = 0 & root.db.d1.s2 = 0, 3, \"ignoreNull\"=\"true\")",
+          "Count_if(root.db.d1.s1 = 1 & root.db.d1.s2 = 0, 3)"
+        };
+    String[] retArray = new String[] {"2,1,"};
+    resultSetEqualTest(
+        "select Count_if(s1=0 & s2=0, 3, 'ignoreNull'='true'), Count_if(s1=1 & s2=0, 3) from root.db.d1",
+        expectedHeader,
+        retArray);
+
+    // keep >= threshold
+    expectedHeader =
+        new String[] {
+          "Count_if(root.db.d1.s1 = 0 & root.db.d1.s2 = 0, keep >= 3, \"ignoreNull\"=\"true\")",
+          "Count_if(root.db.d1.s1 = 1 & root.db.d1.s2 = 0, keep >= 3)"
+        };
+    resultSetEqualTest(
+        "select Count_if(s1=0 & s2=0, keep>=3, 'ignoreNull'='true'), Count_if(s1=1 & s2=0, keep>=3) from root.db.d1",
+        expectedHeader,
+        retArray);
+
+    // keep < threshold
+    expectedHeader =
+        new String[] {
+          "Count_if(root.db.d1.s1 = 0 & root.db.d1.s2 = 0, keep < 3, \"ignoreNull\"=\"true\")",
+          "Count_if(root.db.d1.s1 = 1 & root.db.d1.s2 = 0, keep < 3)"
+        };
+    retArray = new String[] {"0,0,"};
+    resultSetEqualTest(
+        "select Count_if(s1=0 & s2=0, keep<3, 'ignoreNull'='true'), Count_if(s1=1 & s2=0, keep<3) from root.db.d1",
+        expectedHeader,
+        retArray);
+  }
+
+  @Test
+  public void testCountIfRespectNull() {
+    // threshold constant
+    String[] expectedHeader =
+        new String[] {
+          "Count_if(root.db.d1.s1 = 0 & root.db.d1.s2 = 0, 3, \"ignoreNull\"=\"false\")",
+          "Count_if(root.db.d1.s1 = 1 & root.db.d1.s2 = 0, 3, \"ignoreNull\"=\"false\")"
+        };
+    String[] retArray = new String[] {"1,1,"};
+    resultSetEqualTest(
+        "select Count_if(s1=0 & s2=0, 3, 'ignoreNull'='false'), Count_if(s1=1 & s2=0, 3, 'ignoreNull'='false') from root.db.d1",
+        expectedHeader,
+        retArray);
+
+    // keep >= threshold
+    expectedHeader =
+        new String[] {
+          "Count_if(root.db.d1.s1 = 0 & root.db.d1.s2 = 0, keep >= 3, \"ignoreNull\"=\"false\")",
+          "Count_if(root.db.d1.s1 = 1 & root.db.d1.s2 = 0, keep >= 3, \"ignoreNull\"=\"false\")"
+        };
+    resultSetEqualTest(
+        "select Count_if(s1=0 & s2=0, keep>=3, 'ignoreNull'='false'), Count_if(s1=1 & s2=0, keep>=3, 'ignoreNull'='false') from root.db.d1",
+        expectedHeader,
+        retArray);
+
+    // keep < threshold
+    expectedHeader =
+        new String[] {
+          "Count_if(root.db.d1.s1 = 0 & root.db.d1.s2 = 0, keep < 3, \"ignoreNull\"=\"false\")",
+          "Count_if(root.db.d1.s1 = 1 & root.db.d1.s2 = 0, keep < 3, \"ignoreNull\"=\"false\")"
+        };
+    retArray = new String[] {"1,0,"};
+    resultSetEqualTest(
+        "select Count_if(s1=0 & s2=0, keep<3, 'ignoreNull'='false'), Count_if(s1=1 & s2=0, keep<3, 'ignoreNull'='false') from root.db.d1",
+        expectedHeader,
+        retArray);
+  }
+
+  @Test
+  public void testCountIfAlignByDevice() {
```

Comment:
add more than one device. and count_if should take effect in each device instead of among all the devices.